libraries/common/effekt.effekt

```diff
@@ -715,6 +715,13 @@ record Tuple4[A, B, C, D](first: A, second: B, third: C, fourth: D)
 record Tuple5[A, B, C, D, E](first: A, second: B, third: C, fourth: D, fifth: E)
 record Tuple6[A, B, C, D, E, F](first: A, second: B, third: C, fourth: D, fifth: E, sixth: F)
 
+// These
+// =====
+type These[A, B] {
+  This(a: A)
+  That(b: B)
+  These(a: A, b: B)
+}
```

Comment:
Bikeshed: [Unison](https://share.unison-lang.org/@unison/base/code/releases/4.0.0/latest/types/data/OneOrBoth) calls this type `OneOrBoth` with `This | That | Both` which is, I think, a bit cuter.

Reply:
I named it like I know it from haskell - but I really do not care here, if there is a consensus to rename this, I can.

Reply:
Renamed.